app/lib/common/widget/SideMenu.dart

```diff
@@ -88,40 +87,23 @@ class _SideDrawerState extends State<SideDrawer> {
                     username = snapshot.requireData.userId();
                     return GestureDetector(
                       onTap: () {
-                        Navigator.pushNamed(context, '/profile');
+                        Navigator.pushNamed(
+                          context,
+                          '/profile',
+                          arguments: snapshot.requireData,
+                        );
                       },
                       child: Row(
                         children: [
-                          FutureBuilder<List<int>>(
-                            future:
-                                avatar, // a previously-obtained Future<String> or null
-                            builder: (
-                              BuildContext context,
-                              AsyncSnapshot<List<int>> snapshot,
-                            ) {
-                              if (snapshot.hasData) {
-                                return Container(
-                                  margin: EdgeInsets.all(10),
-                                  child: CircleAvatar(
-                                    backgroundImage: MemoryImage(
-                                      Uint8List.fromList(
-                                        snapshot.requireData,
-                                      ),
-                                    ),
-                                    radius: 24,
-                                  ),
-                                );
-                              } else {
-                                return Container(
-                                  margin: EdgeInsets.all(10),
-                                  child: CircleAvatar(
-                                    backgroundColor: Colors.brown.shade800,
-                                    child: const Text('G'),
-                                    radius: 24,
-                                  ),
-                                );
-                              }
-                            },
+                          Container(
+                            margin: EdgeInsets.all(10),
+                            child: CustomAvatar(
+                              radius: 24,
+                              avatar: avatar,
```

Comment:
here it is a bit more hidden as it happens above in L85, but it does it again.

More notably, the list conversion probably isn't even really needed but instead another feature might be implemented on the SDK to directly support the pattern more clean and in that case it'd be nice if we'd only have to change the component, rather than its interface.